pages/api/wishlist/[wineId].js

```diff
@@ -0,0 +1,44 @@
+// add wine to wishlist (POST) or remove wine from wishlist (DELETE)
+
+import dbConnect from "@/db/connect";
+import Wishlist from "@/db/models/Wishlist";
+
+export default async function handler(req, res) {
+  await dbConnect();
+
+  const { wineId } = req.query;
+
+  if (req.method === "POST") {
+    try {
+      // check if already in wishlist
+      const existing = await Wishlist.findOne({ wineId });
+
+      if (existing) {
+        return res.status(400).json({ error: "Wine already in wishlist" });
+      }
+
+      const newWishlistItem = new Wishlist({ wineId });
+      await newWishlistItem.save();
```

Comment:
```suggestion
      await Wishlist.create({ wineId });
```
`create()` runs `new` + `save` in one go. Clean, clear, does the job.